internal/pkg/gateway/api_test.go

```diff
@@ -762,6 +767,25 @@ func TestEndorse(t *testing.T) {
 					Message: "ProposalResponsePayloads do not match",
 				},
 			},
+			postSetup: func(t *testing.T, def *preparedTest) {
+				flogging.ActivateSpec("debug")
+				logObserver := &mock.Observer{}
+				logObserver.WriteEntryStub = func(entry zapcore.Entry, fields []zapcore.Field) {
+					if strings.HasPrefix(entry.Message, "Proposal response mismatch") {
+						for _, field := range fields {
+							def.logFields = append(def.logFields, field.String)
+						}
+					}
+				}
+				flogging.SetObserver(logObserver)
+			},
+			postTest: func(t *testing.T, def *preparedTest) {
+				require.Equal(t, "chaincode response mismatch", def.logFields[0])
+				require.Equal(t, "status: 200, message: , payload: different_response", def.logFields[1])
+				require.Equal(t, "status: 200, message: , payload: mock_response", def.logFields[2])
+				flogging.ActivateSpec("info")
```

Comment:
nit, but - shouldn't you sample what was the spec before and save it rather then defaulting to what you think the spec was before?